nucliadb_texts/src/utils.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2021 Bosutech XXI S.L.
+//
+// nucliadb is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at info@nuclia.com.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+use nucliadb_core::protos::stream_filter::Conjunction;
+use tantivy::query::Occur;
+
+pub fn conjunction_to_occur(conjunction: Conjunction) -> Occur {
+    match conjunction {
+        Conjunction::And => Occur::Must,
+        Conjunction::Or => Occur::Should,
+        Conjunction::Not => Occur::MustNot,
+    }
+}
```

Comment:
What do you think about something like this?
```rust
trait IntoOccur {
    fn into_occur(self) -> Occur;
}

impl IntoOccur for Conjunction {
    fn into_occur(self) -> Occur {
        match self {
            Conjunction::And => Occur::Must,
            Conjunction::Or => Occur::Should,
            Conjunction::Not => Occur::MustNot,
        }
    }
}
```

Reply:
This is seems as a good approach. I do prefer to define this kind of traits for the target type in the following way:
```rust
pub trait OccurParser {
    type Src;
    fn parse(x: Self::Src) -> Occur;
}

impl OccurParser for Occur {
    type Src = Conjunction;
    fn parse(x: Self::Src) -> Occur {
        match x {
            Conjunction::And => Occur::Must,
            Conjunction::Or => Occur::Should,
            Conjunction::Not => Occur::MustNot,
        }
    }
}
```
because then you have a `Into` kind of implementation for free like this:
```rust
trait IntoOccur {
    fn into_occur(self) -> Occur;
}
impl<T> IntoOccur for T
where Occur: OccurParser<Src = T>
{
    fn into_occur(self) -> Occur {
        Occur::parse(self)
    }
}
```
which is what happens with `From` and `Into` in the standard library.